sdk/core/azure_core_amqp/Cargo.toml

```diff
@@ -29,8 +29,12 @@ serde.workspace = true
 serde_amqp = { workspace = true, optional = true }
 serde_bytes = { workspace = true, optional = true }
 tokio.workspace = true
+tokio-socks = { version = "0.5", optional = true }
+native-tls = { version = "0.2", optional = true }
+tokio-native-tls = { version = "0.3", optional = true }
 tracing.workspace = true
 typespec = { workspace = true, features = ["amqp"] }
+typespec_client_core = { workspace = true }
```

Comment:
Everything in here is in `azure_core`. `azure_*` crates outside of core should not reference `typespec` crates.

Reply:
@heaths I referenced it in sdk/eventhubs/azure_messaging_eventhubs/Cargo.toml in order to use the sanitize. any suggestion about that? remove sanitize? leave it in the file? what is the best course of action here?

Reply:
I'm saying that the sanitization function should be exported in `azure_core` as well, in the same module (hopefully, unless we missed something). For any `azure_*` crates we're trying to "hide" the typespec crates as an implementation detail. Everything they export is imported and re-exported by `azure_core` in the same module path. A couple types we "override", but internally make convertible to the types they "override" in typespec.